index.js

```diff
@@ -32,28 +32,72 @@ const fastifyMail = async (fastify, opts) => {
 
   // mail decorator configurations:
   const mail = {
-    createMessage: async function (recipients, templates, context) {
+    // createMessage: async function (recipients, templates, context, opts = {}) {
+    createMessage: async function (message, templatePath, context) {
+      // renders a template with the given context based on the templateName which
+      // should be found in the path provided by templates. Returns "" if the promise is rejected.
+      const renderTemplate = async (templateName) => {
+        return await fastify[propertyName](join(templatePath, templateName), context)
+          .catch(() => { return null })
+      }
+
+      // if opts.templatePath is a string, render the template
+
+      // if from or subject is provided in opts these will be used in preference to templates
+      const from = message.from
+      const to = message.to
+      const subject = message.subject
+      const replyTo = message.replyTo
+      const cc = message.cc
+      const bcc = message.bcc
+      const html = message.html
+      const text = message.text
+
       const [
-        html,
-        subject,
-        from
+        renderedHtml,
+        renderedText
       ] = await Promise.all([
-        fastify[propertyName](join(templates, 'html'), context),
-        fastify[propertyName](join(templates, 'subject'), context),
-        fastify[propertyName](join(templates, 'from'), context)
+        await renderTemplate('html'),
+        await renderTemplate('text')
       ])
 
       return {
-        to: recipients,
         from,
-        subject,
-        html
+        to,
+        cc: cc,
+        bcc: bcc,
+        replyTo,
+        subject: subject,
+        html: renderedHtml || html,
+        text: renderedText || text
       }
     },
-    sendMail: async function (recipients, templates, context) {
+    validateMessage: function (message) {
+      let errors = []
+      errors.pop('test')
+      if (message) {
+        if (!message.to) { errors.push('"to" must be defined') }
```

Comment:
newlines for each of these conditional blocks within, and then some whitespace to separate them

Reply:
done - although i think it's an improvement. The one liners were easy to scan and easy to understand (IMO) as they're all essentially the same and simply using the .push array method that the reader needs to know nothing more about. The nested ifs with extra indentation makes the eyes skip unnecessarily.